src/aurora-dsql-mcp-server/awslabs/aurora_dsql_mcp_server/server.py

```diff
@@ -0,0 +1,316 @@
+        raise e
+
+
+class NoOpCtx:
+    """A No-op context class for error handling in MCP tools."""
+
+    async def error(self, message):
+        """Do nothing.
+
+        Args:
+            message: The error message
+        """
+
+
+async def get_password_token():  # noqa: D103
+    # Generate a fresh password token for each connection, to ensure the token is not expired
+    # when the connection is established
+    client = boto3.client('dsql', region_name=region)
+
+    if database_user == 'admin':
+        return client.generate_db_connect_admin_auth_token(cluster_endpoint, region)
+    else:
+        return client.generate_db_connect_auth_token(cluster_endpoint, region)
+
+
+async def create_connection(ctx):  # noqa: D103
+    password_token = await get_password_token()
+
+    conn_params = {
+        'dbname': DSQL_DB_NAME,
+        'user': database_user,
+        'host': cluster_endpoint,
+        'port': DSQL_DB_PORT,
+        'password': password_token,
+        'application_name': DSQL_MCP_SERVER_APPLICATION_NAME,
+        'sslmode': 'require',
+    }
+
+    logger.info(f'Trying to create connection to {cluster_endpoint} as user {database_user}')
+    # Make a connection to the cluster
+    try:
+        conn = await psycopg.AsyncConnection.connect(**conn_params, autocommit=True)
+    except Exception as e:
+        logger.error(f'{ERROR_CREATE_CONNECTION} : {e}')
+        await ctx.error(f'{ERROR_CREATE_CONNECTION} : {e}')
+        raise e
+
+    return conn
+
+
+async def execute_query(ctx, conn_to_use, query: str, params=None) -> List[dict]:  # noqa: D103
+    if conn_to_use is None:
+        conn = await create_connection(ctx)
+    else:
+        conn = conn_to_use
+
+    try:
+        async with conn.cursor(row_factory=psycopg.rows.dict_row) as cur:
+            await cur.execute(query, params)
+            if cur.rownumber is None:
+                return []
+            else:
+                return await cur.fetchall()
+    except Exception as e:
+        logger.error(f'{ERROR_EXECUTE_QUERY} : {e}')
+        await ctx.error(f'{ERROR_EXECUTE_QUERY} : {e}')
+        raise e
+    finally:
+        if conn_to_use is None:
+            await conn.close()
+
+
+def main():
+    """Run the MCP server with CLI argument support."""
+    parser = argparse.ArgumentParser(
+        description='An AWS Labs Model Context Protocol (MCP) server for Aurora DSQL'
+    )
+    parser.add_argument('--sse', action='store_true', help='Use SSE transport')
+    parser.add_argument('--port', type=int, default=8888, help='Port to run the server on')
+    parser.add_argument(
+        '--cluster_endpoint', required=True, help='Endpoint for your Aurora DSQL cluster'
+    )
+    parser.add_argument('--database_user', required=True, help='Database username')
+    parser.add_argument(
+        '--region',
+        required=True,
+        default='us-west-2',
+        help='AWS region for Aurora DSQL Cluster (default: us-west-2)',
```

Comment:
Would recommend not having a default and letting boto default based on AWS_REGION